server/presence.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"github.com/mattermost/mattermost-plugin-msteams/server/msteams/clientmodels"
+)
+
+const (
+	PresenceActivityAvailable               = "Available"
+	PresenceActivityAway                    = "Away"
+	PresenceActivityBeRightBack             = "BeRightBack"
+	PresenceActivityBusy                    = "Busy"
+	PresenceActivityDoNotDisturb            = "DoNotDisturb"
+	PresenceActivityInACall                 = "InACall"
+	PresenceActivityInAConferenceCall       = "InAConferenceCall"
+	PresenceActivityInactive                = "Inactive"
+	PresenceActivityInAMeeting              = "InAMeeting"
+	PresenceActivityOffline                 = "Offline"
+	PresenceActivityOffWork                 = "OffWork"
+	PresenceActivityOutOfOffice             = "OutOfOffice"
+	PresenceActivityPresenceUnknown         = "PresenceUnknown"
+	PresenceActivityPresenting              = "Presenting"
+	PresenceActivityUrgentInterruptionsOnly = "UrgentInterruptionsOnly"
+
+	PresenceAvailabilityAvailable       = "Available"
+	PresenceAvailabilityAvailableIdle   = "AvailableIdle"
+	PresenceAvailabilityAway            = "Away"
+	PresenceAvailabilityBeRightBack     = "BeRightBack"
+	PresenceAvailabilityBusy            = "Busy"
+	PresenceAvailabilityBusyIdle        = "BusyIdle"
+	PresenceAvailabilityDoNotDisturb    = "DoNotDisturb"
+	PresenceAvailabilityOffline         = "Offline"
+	PresenceAvailabilityPresenceUnknown = "PresenceUnknown"
+)
+
+// userPresenceIsActive returns true if the user is considered online in Teams.
+func userPresenceIsActive(presence *clientmodels.Presence) bool {
+	// If we're missing presence, default to the user being inactive.
+	if presence == nil {
+		return false
+	}
+
+	// Explicitly handle known activity states for being inactive or away.
+	switch presence.Activity {
+	case PresenceActivityOffline, PresenceActivityInactive, PresenceAvailabilityAway:
```

Comment:
For now, I'm proposing we key off organic states -- if a user explicitly sets one of the other configurable states in Teams, we'll treat them as online and managing their notifications there. Experimentation will be required to determine if this matches users behaviour, but the happy path is covered.